tf_agents/agents/dqn/examples/v2/train_eval.py

```diff
@@ -160,7 +165,20 @@ def train_eval(
           output_fc_layer_params,
           num_actions)
     else:
-      q_net = create_feedforward_network(fc_layer_params, num_actions)
+      observation_spec = tf_env.time_step_spec().observation
+      action_spec = tf_env.action_spec()
+      if network_type == 'q_network':
+        q_net = create_q_network(observation_spec,
+                                 action_spec,
+                                 fc_layer_params)
+      elif network_type == 'duel_q_network':
```

Comment:
Move this outside.

Reply:
You mean add a separate train_eval file to support using Q network and duel Q network? So we do not need to modify the original train_eval file?

Reply:
I meant as an auxiliary function within the same file but outside of the train_eval()

Ex:

```
def q_network(observation_spec, action_spec, fc_layer_params=(100,)):
....

def dual_q_network(observation_spec, action_spec, fc_layer_params=(100,), a_fc_layer_params=(50,),
    v_fc_layer_params=(50,)):
   ....

```